ChallengesWithTestsMarkVII/ChallengesSet05.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Text;
+
+namespace ChallengesWithTestsMarkVII
+{
+    public class ChallengesSet05
+    {
+        public int GetNextNumberDivisibleByN(int startNumber, int n)
+        {
+            if (startNumber == 0)
+            {
+                return n;
+            }
+
+            List<int> firstlist = new List<int>();
+            int maxLoop = startNumber * n;
+            int[] numbersDivisblebyN = new int[0];
+
+            for (int i = 0; i < maxLoop; i++)
+            {
+                if (i > startNumber && i % n == 0)
+                {
+                    firstlist.Add(i);
+                }
+
+            }
+
+            firstlist.ToArray();
+            return firstlist[0];
+
+        }
+
+        public void ChangeNamesOfBusinessesWithNoRevenueTo_CLOSED(Business[] businesses)
+        {
+            for (int i = 0; i < businesses.Length; i++)             {                 if (businesses[i].TotalRevenue == 0 || businesses[i].TotalRevenue == null)                 {                     businesses[i].Name = "CLOSED";                 }             }         }
+
+        public bool IsAscendingOrder(int[] numbers)
+        {
+            if (numbers == null || numbers.Length == 0)
+            {
+                return false;
+            }
+
+            int NumbersOutofOrderCount = 0;
+            int currentNumber = 0;
+            int nextNumber = 0;
+
+            for (int i = 0; i < numbers.Length - 1; i++)
+            {
+                currentNumber = numbers[i];
+                nextNumber = numbers[i + 1];
+                if (currentNumber > nextNumber)
+                {
+                    NumbersOutofOrderCount += 1;
+                }
+            }
+
+            if (NumbersOutofOrderCount >= 1)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        public int SumElementsThatFollowAnEven(int[] numbers)
+        {
+            if (numbers == null)
+            {
+                return 0;
+            }
+
+            int sum = 0;
+            int currentNumber = 0;
+            int nextNumber = 0;
```

Comment:
This doesn't need to be declared this far in advance (side note: different developers have different opinions on where variables should be declared, so some people won't agree with me on this). It's only used within the scope of the for-loop and could be declared on line 85